### PR TITLE
EOS:16437 - Add init scetion to new python s3_setup script

### PR DESCRIPTION
**Pull request by amitwac2608**
Signed-off-by: Amit Kumar <amit.kumar@seagate.com>

-----
[View rendered scripts/provisioning/s3setup/README.md](https://github.com/Seagate/cortx-s3server/blob/br/ak/EOS-16437-s3setup-init/scripts/provisioning/s3setup/README.md)

### CODE REVIEW

File: installhelper.sh [hunk 1]
```diff
@@ -108,14 +108,15 @@ cp scripts/provisioning/cleanupcmd.py $S3_INSTALL_LOCATION/bin
...
+cp scripts/ldap/ldapaccountaction.py $S3_INSTALL_LOCATION/bin
 
 # Copy the mini-provisioner config files
 cp scripts/provisioning/s3setup_prereqs.json $S3_MINI_PROV_CFG_LOCATION/
 cp scripts/provisioning/s3_prov_config.yaml $S3_MINI_PROV_CFG_LOCATION/
 
 # Copy the S3 reset scripts
 cp scripts/reset/* $S3_INSTALL_LOCATION/reset
-cp scripts/provisioning/s3setup/clean_open_ldap_by_s3.sh $S3_INSTALL_LOCATION/reset
+cp scripts/provisioning/clean_openldap.sh $S3_INSTALL_LOCATION/reset
```
Comment (nileshgovande) [inline installhelper.sh:119]: @SaumitraK531 - Make a JIRA to clean up

File: scripts/provisioning/cleanupcmd.py [hunk 2]
```diff
@@ -115,31 +112,32 @@ def delete_ldap_account(self, ldap_acc: str):
       s3cipher_obj = CortxS3Cipher(None, True, 22, acc_attributes_dict["cipherConstKey"])
```
Comment (nileshgovande) [inline scripts/provisioning/cleanupcmd.py:112]: Should not need to use cipher.
LDAPsearch should retrive the same - investigate

File: scripts/provisioning/postinstallcmd.py [hunk 3]
```diff
@@ -41,21 +40,13 @@ def process(self):
     """Main processing function."""
     sys.stdout.write(f"Processing {self.name}\n")
```
Comment (nileshgovande) [inline scripts/provisioning/postinstallcmd.py:41]: Running Validations

File: scripts/provisioning/s3_setup [hunk 4]
```diff
@@ -231,37 +231,20 @@ create_s3backgroundelete_keys()
...
-    fi
+    create_s3backgroundelete_keys
+
+    # TODO - use s3confstore to set the access_key and secret_key in confstore.
+    # once the s3backgrounddelete is updated to read the keys from confstore, instead of config.yaml file
+    sed -i "s/background_account_access_key:.*$/background_account_access_key: \"$access_key\"/g" /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml
+    sed -i "s/background_account_secret_key:.*$/background_account_secret_key: \"$secret_key\"/g" /opt/seagate/cortx/s3/s3backgrounddelete/config.yaml
+    sed -i "s/ak:.*/ak: $access_key/g" /opt/seagate/cortx/s3/install/ldap/background_delete_account.ldif
+    sed -i "s/sk:.*/sk: $secret_key/g" /opt/seagate/cortx/s3/install/ldap/background_delete_account.ldif
+    sed -i "s/ak=.*/ak=$access_key,ou=accesskeys,dc=s3,dc=seagate,dc=com/g" /opt/seagate/cortx/s3/install/ldap/background_delete_account.ldif
```
Comment (nileshgovande) [inline scripts/provisioning/s3_setup:245]: 1)Try deleting this block of code and 
2)Try deleting ldif files as well.
3)Try deleting the 'access-key' and 'secret-key' parameters from bgdel config files

**Review by nileshgovande — APPROVED** [review]

**Review by AjinkyaDhumal — APPROVED** [review]

**Pull request merged**